extension/mediator.js

```diff
@@ -140,13 +143,15 @@ class Mediator {
         switch (message.command) {
             case "translate":
                 if (this.isMainFrame) {
+                    if (!("origin" in message)) message.origin = THIS_ORIGIN;
```

Comment:
Is it possible to make this unconditional, by putting it in the right place where the origin of a message is unambiguous?

Reply:
Actually, it shouldn't be conditional, fixed. I think it's the right place, otherwise, we'll need to add this to both InPageTranslation and OutboundTranslation. I prefer to keep this logic in mediator.